pages/api/events-count.ts

```diff
@@ -0,0 +1,42 @@
+import { ObjectId } from 'mongodb';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { getSession } from 'next-auth/react';
+import dbConnect from '../../middleware/database';
+import Event from '../../models/event';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse<any>) {
+	const session = await getSession({ req });
+	if (session?.userType !== 'ORGANIZER') return res.status(403).send('Forbidden');
+
+	await dbConnect();
+	switch (req.method) {
+		case 'GET':
+			const count = await Event.aggregate([
+				{
+					$lookup: {
+						from: 'users',
+						localField: '_id',
+						foreignField: 'eventsAttended',
+						as: 'checkIns',
+					},
+				},
+				{
+					$match: {
+						'checkIns.0': {
+							$exists: true,
+						},
+					},
+				},
+				{
+					$project: {
+						count: {
+							$size: '$checkIns',
+						},
+					},
+				},
+			]);
```

Comment:
Can you add some comments to explain how this works? This looks like one of our most complex queries lol